### PR TITLE
[Recorder] Add multiout equivalent in transLayer @open sesame 06/09 15:12

### PR DESCRIPTION
**Pull request by zhoonit**
- [Recorder] Add multiout equivalent in transLayer

```
When getting gradient from `gradientTape` it adds all derivatives of the
dependent layer. While this behavior is true, we need to measure to what
extent each layer contributed to the layer.

This patch adds a multiout layer which consists of some stublayer that
seperates derivatives one by one.

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

```

### CODE REVIEW

**Review by jijoongmoon — APPROVED** [review]
LGTM

File: test/input_gen/transLayer.py [hunk 1]
```diff
@@ -144,6 +144,36 @@ def to_nntr_weights(self, weights):
...
+    # @param tf_layer tf_layer to get number of outputs
+    # @param num_output explicit number to generate number of output
+    def __init__(self, tf_layer = None, *args, num_output, **kwargs):
+        if not tf_layer:
+            tf_layer = K.layers.Layer()
+
+        super().__init__(tf_layer, *args, **kwargs)
+
+        # this enables seperating gradient one by one
+        self.stub_layers = [K.layers.Lambda(lambda x: x + 0) for i in range(num_output)]
```
Comment (kparichay) [inline test/input_gen/transLayer.py:164]: How about simply
```suggestion
        self.stub_layers = [K.layers.Lambda(lambda x: x) for i in range(num_output)]
```

Comment (zhoonit) [reply test/input_gen/transLayer.py:164]: Checked this, and it didn't work. I suppose if I don't do x: x + 0, I think tensorflow optimizer optimizes this to a `noop`.

Comment (kparichay) [reply test/input_gen/transLayer.py:164]: ok.

**Review by kparichay — APPROVED** [review]
LGTM

**Pull request merged**